sklearn/datasets/descr/diabetes.rst

```diff
@@ -2,7 +2,37 @@ Diabetes dataset
 
 Notes
 -----
+
 Ten baseline variables, age, sex, body mass index, average blood
 pressure, and six blood serum measurements were obtained for each of n =
 442 diabetes patients, as well as the response of interest, a
 quantitative measure of disease progression one year after baseline.
+
+Note that the 10 x variables have been standardized to have mean 0 and squared length = 1 (sum(x^2)=1).
```

Comment:
From the data, it really does appear to be column-based normalization, such that the mean of each column is `0.0` and the variance is `1/442` (!).  (Or, in other words, the Euclidean length of each column is `1`.)  I too found this surprising.
```
In [12]: ds = sklearn.datasets.load_diabetes()

In [13]: ds.data.mean(axis=0)
Out[13]: 
array([ -3.63428493e-16,   1.30834257e-16,  -8.04534920e-16,
         1.28165452e-16,  -8.83531559e-17,   1.32702421e-16,
        -4.57464634e-16,   3.77730150e-16,  -3.83085422e-16,
        -3.41288202e-16])

In [14]: ds.data.var(axis=0) * 442
Out[14]: array([ 1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.,  1.])
```